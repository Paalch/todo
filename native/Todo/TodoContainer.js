import React, { Component } from 'react'
import { Content, Divider, List } from 'native-base'
import { AsyncStorage } from 'react-native';
import Todo from './Todo'
import TodoInput from './TodoInput'
import moment from "moment";

export default class TodoContainer extends Component<{}> {
    constructor(props) {
        super(props)

        this.state = {
            todos: []
        }
    }

    componentWillMount = async () => {
        //localStorage.clear()
        console.log("test")
        let localTodos = JSON.parse(await AsyncStorage.getItem('todos'))
        this.setState({
            todos: localTodos || []
        })
    }

    checkBoxClick = (todo) => {
        console.log("im running")
        const { todos } = this.state
        const i = todos.indexOf(todo)
        if (i >= 0) {
            todos[i].checked = !todos[i].checked
            this.updateState({
                todos: todos
            })
        } else {
            console.error(`[TodoContainer](checkBoxClick) Couldn't find object at index ${i}`)
        }
        
    }

    onButtonClick = (text) => {
        const { todos } = this.state
        let todo = {
            text: text,
            date: moment(),
            checked: false,
            isStar:false,
        }
        todos.splice(0,0,todo)
        this.updateState({
            todos: todos
        })
    }

    deleteItem = (todo) => {
        let { todos } = this.state
        const i = todos.indexOf(todo)
        if (i >= 0) {
            todos.splice(i, 1)
            this.updateState({
                todos: todos
            })
        } else {
            console.error(`[TodoContainer](checkBoxClick) Couldn't find object at index ${i}`)
        }
    }

    updateLocalStorage = async () => {
        const { todos } = this.state
        try {
            await AsyncStorage.setItem('todos', JSON.stringify(todos));
        } catch (error) {
            console.error(error)
        }
    }


    updateState = (state) => {
        this.setState(state, () => {
            this.updateLocalStorage()
        })
    }

    updateToDos = (todo) => {
        const {todos} = this.state;
        for(let i = 0; i < todos.length; i++) {
            if(todos[i] === todo){
                todos.splice(i,1)
                todo.isStar ? todos.splice(0,0,todo) :  todos.push(todo)
                this.updateState(todos)
                break
            }
        }
        console.log(todos)
    }


    render() {
        const { todos } = this.state
        return(
                <Content>
                    <TodoInput onButtonClick={this.onButtonClick}/>
                    <List>
                    { todos.map((todo) => <Todo key={todo.date} todo={todo} checkBoxClick={this.checkBoxClick}
                                                deleteItem={this.deleteItem} onButtonClick = {this.onButtonClick}
                                                updateToDos = {this.updateToDos} />) }
                    </List>
                </Content>
        )
    }
}