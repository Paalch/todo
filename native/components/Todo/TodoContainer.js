import React, { Component } from 'react'
import { View, Content, List, Text } from 'native-base'
import { AsyncStorage } from 'react-native';
import Todo from './Todo'
import moment from "moment";
import FABNewItem from '../FABNewItem'
import NewTodoModal from './NewTodoModal'
import { todoCont } from '../../styles'

export default class TodoContainer extends Component {
    constructor(props) {
        super(props)

        //Sets the state
        this.state = {
            newModalOpen: false,
            todos: []
        }
    }

  /**
   * Loads content from localStorage
   */
  componentWillMount = async () => {
        let localTodos = JSON.parse(await AsyncStorage.getItem('todos'))
        this.setState({
            todos: localTodos || []
        })
    }

  /**
   * Handles the checkbox button in todo
   * @param todo
   */
  checkBoxClick = (todo) => {
        const { todos } = this.state
        const i = todos.indexOf(todo)
        if (i >= 0) {
            todos[i].checked = !todos[i].checked
            this.updateState({
                newModalOpen: false,
                todos: todos
            })
        } else {
            console.error(`[TodoContainer](checkBoxClick) Couldn't find object at index ${i}`)
        }

    }

  /**
   * Adds a new todo to the list, and updates localStorage
   * @param text
   */
  newTodo = (text) => {
        const { todos } = this.state
        let todo = {
            text: text,
            date: moment(),
            checked: false,
            isStar:false,
        }
        todos.splice(0,0,todo)
        this.updateState({
            todos: todos
        })
    }

    /**
     * Toggles new note modal on/off
     */
    toggleNewModal = () => {
        this.setState({
            newModalOpen: !this.state.newModalOpen
        })
    }

  /**
   * Removes given todo
   * @param todo
   */
  deleteItem = (todo) => {
        let { todos } = this.state
        const i = todos.indexOf(todo)
        if (i >= 0) {
            todos.splice(i, 1)
            this.updateState({
                todos: todos
            })
        } else {
            console.error(`[TodoContainer](checkBoxClick) Couldn't find object at index ${i}`)
        }
    }

  /**
   * Updates localStorage
   * @returns {Promise.<void>}
   */
  updateLocalStorage = async () => {
        const { todos } = this.state
        try {
            await AsyncStorage.setItem('todos', JSON.stringify(todos));
        } catch (error) {
            console.error(error)
        }
    }

  /**
   * Updates state
   * @param state
   */
  updateState = (state) => {
        this.setState(state, () => {
            this.updateLocalStorage()
        })
    }

  /**
   * Updates the todo list for a given todo
   * @param todo
   */
  updateToDos = (todo) => {
        const {todos} = this.state;
        for(let i = 0; i < todos.length; i++) {
            if(todos[i] === todo){
                todos.splice(i,1)
                todo.isStar ? todos.splice(0,0,todo) :  todos.push(todo)
                this.updateState(todos)
                break
            }
        }
    }

    render() {
        //define constants
        const { todos, newModalOpen } = this.state
        return(
                <View style={todoCont.view}>
                    <Content>

                    <NewTodoModal
                        toggleModal={this.toggleNewModal}
                        onButtonSaveClick={this.newTodo}
                        isOpen={ newModalOpen }/>

                    <List>
                    { todos.map((todo) => (
                        <Todo key={todo.date}
                            todo={todo}
                            checkBoxClick={this.checkBoxClick}
                            deleteItem={this.deleteItem}
                            onButtonClick = {this.newTodo}
                            updateToDos = {this.updateToDos}
                        />)
                    )}
                    </List>

                  { todos.length ?
                    <Text style={todoCont.endText}>
                        End of your list
                    </Text> :
                    <Text style={todoCont.endText}>
                        No todos
                    </Text>
                  }
                    </Content>
                    <View>
                  <FABNewItem toggleModal={this.toggleNewModal}/>
                    </View>

                </View>
        )
    }

}