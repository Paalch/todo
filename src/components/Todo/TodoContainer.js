import React, { Component } from 'react'
import { Container, Divider } from 'semantic-ui-react'

import Todo from './Todo'
import TodoInput from './TodoInput'
import Navbar from '../Navbar'
import Footer from "../Footer";

export default class TodoContainer extends Component {
    constructor(props) {
        super(props)

        this.state = {
            todos: []
        }
    }

    componentWillMount = () => {
        //localStorage.clear()
        let localTodos = JSON.parse(localStorage.getItem('todos'))
        this.setState({
            todos: localTodos || []
        })
    }

    checkBoxClick = (todo) => {
        const { todos } = this.state
        const i = todos.indexOf(todo)
        if (i >= 0) {
            todos[i].checked = !todos[i].checked
            this.updateState({
                todos: todos
            })
        } else {
            console.error(`[TodoContainer](checkBoxClick) Couldn't find object at index ${i}`)
        }
        
    }

    onButtonClick = (text) => {
        const { todos } = this.state
        let todo = {
            text: text,
            date: new Date(),
            checked: false
        }
        todos.push(todo)
        this.updateState({
            todos: todos
        })
    }

    deleteItem = (todo) => {
        let { todos } = this.state
        const i = todos.indexOf(todo)
        if (i >= 0) {
            todos.splice(i, 1)
            this.updateState({
                todos: todos
            })
        } else {
            console.error(`[TodoContainer](checkBoxClick) Couldn't find object at index ${i}`)
        }
    }

    updateLocalStorage = () => {
        const { todos } = this.state
        localStorage.setItem('todos', JSON.stringify(todos))
    }

    updateState = (state) => {
        this.setState(state, () => {
            this.updateLocalStorage()
        })
    }

    render() {
        const { todos } = this.state
        return(
            <div>
                <Navbar />
                <Container text>
                    <Divider hidden />
                    <TodoInput onButtonClick={this.onButtonClick}/>
                    <Divider hidden/>
                    { todos.map((todo) => <Todo key={todo.date} todo={todo} checkBoxClick={this.checkBoxClick} deleteItem={this.deleteItem}/>) }
                </Container>
                <Divider hidden />
                <Footer/>
            </div>
        )
    }
} 