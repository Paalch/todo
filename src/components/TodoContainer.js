import React, { Component } from 'react'
import { Container } from 'semantic-ui-react'

import Todo from './Todo'
import TodoInput from './TodoInput'

export default class TodoContainer extends Component {
    constructor(props) {
        super(props)

        this.state = {
            todos: [
                {text: 'Oppgave i morgen..', checked: false, date: '01-01-1970'}
            ]
        }

        this.checkBoxClick = this.checkBoxClick.bind(this)
        this.onButtonClick = this.onButtonClick.bind(this)
    }

    checkBoxClick() {

    }

    onButtonClick(text) {
        const { todos } = this.state
        let todo = {
            text: text,
            date: new Date(),
            checked: false
        }
        todos.push(todo)
        this.setState({
            todos: todos
        })
    }

    render() {
        const { todos } = this.state
        return(
            <Container text>
                <TodoInput onButtonClick={this.onButtonClick}/>
                { todos.map((todo) => <Todo todo={todo} />) }
            </Container>
        )
    }
} 