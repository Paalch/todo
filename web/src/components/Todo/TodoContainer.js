import React, { Component } from 'react'
import { Container, Divider } from 'semantic-ui-react'
import moment from 'moment'

import Todo from './Todo'
import TodoInput from './TodoInput'
import Navbar from '../Navbar'
import Footer from '../Footer'

export default class TodoContainer extends Component {
    constructor (props) {
        super(props)

        // init state
        this.state = {
            todos: []
        }
    }

    // loads todo from localStorage
    componentWillMount = () => {
        // localStorage.clear()
        let localTodos = JSON.parse(localStorage.getItem('todos'))
        this.setState({
            todos: localTodos || []
        })
    }

    // handles checkbox click
    checkBoxClick = (todo) => {
        const { todos } = this.state
        const i = todos.indexOf(todo)
        if (i >= 0) {
            todos[i].checked = !todos[i].checked
            this.updateState({
                todos: todos
            })
        } else {
            console.error(`[TodoContainer](checkBoxClick) Couldn't find object at index ${i}`)
        }
    }

    // Creates new todo and adds it to array
    newTodo = (text) => {
        const { todos } = this.state
        let todo = {
            text: text,
            date: moment(),
            checked: false,
            isStar: false
        }
        todos.splice(0, 0, todo)
        this.updateState({
            todos: todos
        })
    }

    // removes todo
    deleteItem = (todo) => {
        let { todos } = this.state
        const i = todos.indexOf(todo)
        if (i >= 0) {
            todos.splice(i, 1)
            this.updateState({
                todos: todos
            })
        } else {
            console.error(`[TodoContainer](checkBoxClick) Couldn't find object at index ${i}`)
        }
    }

    // updates localstorage
    updateLocalStorage = () => {
        const { todos } = this.state
        localStorage.setItem('todos', JSON.stringify(todos))
    }

    // Updates state
    updateState = (state) => {
        this.setState(state, () => {
            this.updateLocalStorage()
        })
    }

    // Updates the order of the todos
    updateToDos = (todo) => {
        const {todos} = this.state
        console.log(todos)
        console.log(todo)
        for (let i = 0; i < todos.length; i++) {
            if (todos[i] === todo) {
                console.log('test')
                todos.splice(i, 1)
                todo.isStar ? todos.splice(0, 0, todo) : todos.push(todo)
                this.updateState(todos)
                break
            }
        }
    }

    render () {
        // define constants
        const { todos } = this.state
        return (
            <div>
                <Navbar />
                <Container text>
                    <Divider hidden />
                    <TodoInput onButtonClick={this.newTodo}/>
                    <Divider hidden/>
                    { todos.map((todo) => <Todo key={todo.date} todo={todo} checkBoxClick={this.checkBoxClick} deleteItem={this.deleteItem} onButtonClick = {this.newTodo} updateToDos = {this.updateToDos} />) }
                </Container>
                <Divider hidden />
                <Footer/>
            </div>
        )
    }
}
