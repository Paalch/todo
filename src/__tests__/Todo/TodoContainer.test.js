import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import TodoContainer from '../../components/Todo/TodoContainer';
import Adapter from 'enzyme-adapter-react-15'
import toJson from 'enzyme-to-json'


Enzyme.configure({ adapter: new Adapter() });

test('Todo component should render as expected', () => {
  const component = shallow(<TodoContainer/>);
  console.log(component);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});