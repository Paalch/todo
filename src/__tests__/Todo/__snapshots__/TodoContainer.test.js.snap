// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component should render as expected 1`] = `
<div>
  <Navbar />
  <Container
    text={true}
  >
    <Divider
      hidden={true}
    />
    <TodoInput
      onButtonClick={[Function]}
    />
    <Divider
      hidden={true}
    />
  </Container>
  <Divider
    hidden={true}
  />
  <Footer />
</div>
`;
